### PR TITLE
Rename project to openfaas-operator and other fixes

### LINKED ISSUE #17: Naming of repo
This repo name is going to be confusing.. what else could we come up with that expresses it better?

Yes I know I had something to do with finding the name originally.

### LINKED ISSUE #19: Run unit tests with TravisCI
We should run the unit tests as part of the CI pipeline

### LINKED ISSUE #20: List API call crash when replicas are not set
When the replicas field has no value the `/list` API call fails.

### PR DESCRIPTION
**Pull request by stefanprodan**
This PR does the following:
* rename packages and image to openfaas-operator Fix #17 #18 
* run unit tests as part of the CI pipeline Fix #19 
* allows HPA v2 to take control of the auto-scaling (replicas must not be set in CR)
* change list API to return the deployment desired replica count #20 

`stefanprodan/openfaas-operator:0.6.0` was tested on GKE 1.10 with HPA v2 enabled.

### CODE REVIEW

**Comment by stefanprodan** [comment]
@alexellis before merging please create `functions/openfaas-operator` on DockerHub and afterwords make a `0.6.0` GitHub release.

**Comment by alexellis** [comment]
@stefanprodan I'm not sure about "please create functions/openfaas-operator on DockerHub"?

If we push an image even if it's non existing DockerHub will create it.

File: pkg/server/list.go [hunk 1]
```diff
@@ -23,9 +23,13 @@ func makeListHandler(namespace string, client clientset.Interface) http.HandlerF
 		}
 
 		for _, item := range res.Items {
+			replicas := uint64(1)
```
Comment (alexellis) [inline pkg/server/list.go]: How do we get into this scenario where `replicas` is nil?

Comment (stefanprodan) [reply pkg/server/list.go]: I've explain in https://github.com/openfaas-incubator/openfaas-operator/issues/20

**Pull request merged**